eng/common/TestResources/build-test-resource-config.yml

```diff
@@ -44,3 +47,16 @@ steps:
         displayName: Merge Test Resource Configurations
         ${{ if parameters.EnvVars }}:
           env: ${{ parameters.EnvVars }}
+
+  - ${{ if ne(parameters.SubscriptionConfigurationFilePath, '') }}:
```

Comment:
Doesn't this also need to exist in the each loop? We may have multiple file configs for each cloud/env the same as we do for keyvault based configs.

Also would it be simpler to empty string check the `SubscriptionConfigurationFilePath` parameter within the existing script blocks instead of adding an additional step?

Reply:
I'm wondering similar should we do all these in one step? Even collapsing the foreach loop and just doing that inline with the correct amount of loggin?

Reply:
My inclination is to collapse these into a single task.

Reply:
Do it!

Reply:
Expanding a macro-syntax variable expression in this way gives unfortunate results:

https://dev.azure.com/azure-sdk/playground/_build/results?buildId=3874379&view=logs&j=5c65763d-3a87-52b5-f04f-2257d72cd1d6&t=9e2ea864-8b34-535b-2546-eff221fc97e7

```
[
  "{ "hello": "world1" }",
  "{ "hello": "world2" }"
]
```

Notice the wrapping quotes around the JSON expression contained within. 

This could be fixed with a lot of fragile parsing or, expanded into tasks as is already the case. 

Since the file list is only strings this can be done easily with `convertToJson` for a list of strings... but it cannot be easily done for a list of macro-syntax JSON expressions.

Reply:
Does that blow up if you directly pass it to ConvertFrom-Json? What about a simple string replace `"{` with `{` and `}"` with `}`?

Reply:
Found a way to get this working.